server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/WebPushContract.scala

```diff
@@ -0,0 +1,508 @@
+           |        [
+           |            "PushSubscription/set",
+           |            {
+           |                "updated": {
+           |                    "$pushSubscriptionId": {}
+           |                }
+           |            },
+           |            "c1"
+           |        ]
+           |    ]
+           |}""".stripMargin)
+
+    // GIVEN 8 days passes
+    clock.setInstant(clock.instant().plus(8, ChronoUnit.DAYS))
+
+    // WHEN bob receives a mail
+    sendEmailToBob(server)
+
+    // THEN bob has a stateChange on the push gateway
+    TimeUnit.MILLISECONDS.sleep(200)
+
+    pushServer.verify(HttpRequest.request()
+      .withPath(PUSH_URL_PATH)
+      .withBody(json(
+        s"""{
+           |    "@type": "StateChange",
+           |    "changed": {
+           |        "$ACCOUNT_ID": "$${json-unit.ignore-element}"
+           |    }
+           |}""".stripMargin)),
+      VerificationTimes.exactly(0))
+  }
+
+  @Test
+  def webPushShouldNotPushToPushServerWhenDeletedSubscription(server: GuiceJamesServer, pushServer: ClientAndServer): Unit = {
+    // Setup mock-server for callback
+    val bodyRequestOnPushServer: AtomicReference[String] = setupPushServerCallback(pushServer)
+
+    // WHEN bob creates a push subscription
+    val pushSubscriptionId: String = createPushSubscription(pushServer)
+    // THEN a validation code is sent
+    awaitAtMostTenSeconds.untilAsserted { () =>
+      pushServer.verify(HttpRequest.request()
+        .withPath(PUSH_URL_PATH)
+        .withBody(json(
+          s"""{
+             |    "@type": "PushVerification",
+             |    "pushSubscriptionId": "$pushSubscriptionId",
+             |    "verificationCode": "$${json-unit.any-string}"
+             |}""".stripMargin)),
+        VerificationTimes.atLeast(1))
+    }
+
+    // GIVEN bob retrieves the validation code from the mock server
+    val verificationCode: String = Json.parse(bodyRequestOnPushServer.get()).asInstanceOf[JsObject]
+      .value("verificationCode")
+      .asInstanceOf[JsString]
+      .value
+
+    // WHEN bob updates the validation code via JMAP
+    val updateVerificationCodeResponse: String = updateValidateVerificationCode(pushSubscriptionId, verificationCode)
+
+    // THEN  it succeed
+    assertThatJson(updateVerificationCodeResponse)
+      .isEqualTo(
+        s"""{
+           |    "sessionState": "${SESSION_STATE.value}",
+           |    "methodResponses": [
+           |        [
+           |            "PushSubscription/set",
+           |            {
+           |                "updated": {
+           |                    "$pushSubscriptionId": {}
+           |                }
+           |            },
+           |            "c1"
+           |        ]
+           |    ]
+           |}""".stripMargin)
+
+    // GIVEN bob deletes the push subscription
+    val pushSubscriptionProbe: PushSubscriptionProbe = server.getProbe(classOf[PushSubscriptionProbe])
```

Comment:
Don't forget to use the revoke method instead of the probe (and to remove the revoke method you created on the probe) when it's merged : https://github.com/apache/james-project/pull/726

It's stable and approved, just need a green build now to merge it...